pkg/services/ngalert/notifier/channels/sensugo.go

```diff
@@ -0,0 +1,201 @@
+	}
+
+	return &SensuGoNotifier{
+		NotifierBase: old_notifiers.NewNotifierBase(model),
+		URL:          url,
+		Entity:       model.Settings.Get("entity").MustString(),
+		Check:        model.Settings.Get("check").MustString(),
+		Namespace:    model.Settings.Get("namespace").MustString(),
+		Handler:      model.Settings.Get("handler").MustString(),
+		APIKey:       apikey,
+		Message:      model.Settings.Get("message").MustString(`{{ template "default.message" .}}`),
+		log:          log.New("alerting.notifier.slack"),
+		tmpl:         t,
+	}, nil
+}
+
+// Notify send alert notification to Sensu Go
+func (sn *SensuGoNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	sn.log.Info("Sending Sensu Go result")
+
+	data := notify.GetTemplateData(ctx, sn.tmpl, as, gokit_log.NewNopLogger())
+	var tmplErr error
+	tmpl := notify.TmplText(sn.tmpl, data, &tmplErr)
+
+	ruleUIDs := make(map[string]struct{})
+	ruleNames := make(map[string]struct{})
+	for _, a := range as {
+		v, ok := a.Labels[ngmodels.UIDLabel]
+		if ok {
+			_, exists := ruleUIDs[string(v)]
+			if !exists {
+				ruleUIDs[string(v)] = struct{}{}
+			}
+		}
+		v, ok = a.Labels[model.AlertNameLabel]
+		if ok {
+			_, exists := ruleNames[string(v)]
+			if !exists {
+				ruleNames[string(v)] = struct{}{}
+			}
+		}
+	}
+
+	var buffer bytes.Buffer
+	for k := range ruleUIDs {
+		buffer.WriteString(fmt.Sprintf("%s,", k))
+	}
+	strRuleUIDs := strings.TrimRight(buffer.String(), ",")
+
+	buffer.Reset()
+	for k := range ruleNames {
+		buffer.WriteString(fmt.Sprintf("%s,", k))
+	}
+	strRuleNames := strings.TrimRight(buffer.String(), ",")
+
+	// Sensu Go alerts require an entity and a check. We set it to the user-specified
+	// value (optional), else we fallback and use the grafana rule anme  and ruleID.
+	entity := sn.Entity
+	if entity == "" {
+		if ruleNames != nil {
+			// Sensu Go alerts cannot have spaces in them
+			entity = strings.ReplaceAll(strRuleNames, " ", "_")
+		} else {
+			entity = "default"
+		}
+	}
+
+	check := sn.Check
+	if check == "" {
+		if strRuleUIDs != "" {
+			check = strRuleUIDs
+		} else {
+			check = "default"
+		}
+	}
+
+	alerts := types.Alerts(as...)
+	status := 0
+	if alerts.Status() == model.AlertFiring {
+		// TODO figure out about NoData old state (we used to send status 1 in that case)
+		status = 2
+	}
+
+	namespace := sn.Namespace
+	if namespace == "" {
+		namespace = "default"
+	}
+
+	var handlers []string
+	if sn.Handler != "" {
+		handlers = []string{sn.Handler}
+	}
+
+	ruleURL := path.Join(sn.tmpl.ExternalURL.String(), "/alerting/list")
+	bodyMsgType := map[string]interface{}{
+		"entity": map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"name":      entity,
+				"namespace": namespace,
+			},
+		},
+		"check": map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"name": check,
+				"labels": map[string]string{
+					"ruleName": strRuleNames,
+					"ruleUId":  strRuleUIDs,
```

Comment:
Given we will have these for individual alerts in `output`, do we require it here?

Reply:
I have changed `output` and I set it to "default" if it's missing. But I don't know why we used to pass the `ruleId` label here. My concern initially was that not passing it could break existing integrations but changing the property would do so as well. So I'm a bit indecisive: should I completely removed it or keep passing this information to the old property `ruleId`? Also WDYT about the `output`? Should I set the rule UIDs or leave it `default`?

Reply:
> My concern initially was that not passing it could break existing integrations but changing the property would do so as well

Fair enough, I dont know if things would break if we dont have it. But since we are having multiple IDs now, it might definitely break in some way. I would prefer we remove the field, because its always easier to add more fields than remove without breaking.

> Also WDYT about the output?

This looks fine. Originally it was message too I believe.

> Should I set the rule UIDs or leave it default?

So hard to decide! I would say lets remove it?